quickjs.c

```diff
@@ -6717,23 +6735,24 @@ static void build_backtrace(JSContext *ctx, JSValue error_obj, JSValue filter_fu
             dbuf_printf(&dbuf, "    at %s", str1);
             JS_FreeCString(ctx, func_name_str);
 
-            if (b) {
+            if (b && sf->cur_pc) {
                 const char *atom_str;
                 int line_num1, col_num1;
+                uint32_t pc;
 
-                /* Bytecode functions must have cur_pc set in the stack frame. */
-                if (sf->cur_pc == NULL)
-                    abort();
-
-                line_num1 = find_line_num(ctx, b,
-                                          sf->cur_pc - b->byte_code_buf - 1,
-                                          &col_num1);
+                pc = sf->cur_pc - b->byte_code_buf - 1;
+                line_num1 = find_line_num(ctx, b, pc, &col_num1);
                 atom_str = b->filename ? JS_AtomToCString(ctx, b->filename) : NULL;
                 dbuf_printf(&dbuf, " (%s", atom_str ? atom_str : "<null>");
                 JS_FreeCString(ctx, atom_str);
                 if (line_num1 != -1)
                     dbuf_printf(&dbuf, ":%d:%d", line_num1, col_num1);
                 dbuf_putc(&dbuf, ')');
+            } else if (b) {
+                // FIXME(bnoordhuis) Missing `sf->cur_pc = pc` in bytecode
+                // handler in JS_CallInternal. Almost never user observable
+                // except with intercepting JS proxies that throw exceptions.
+                dbuf_printf(&dbuf, " (missing)");
```

Comment:
@saghul :point_up: is a pre-existing bug that is more visible now

An example is:

1. the OP_define_field handler in JS_CallInternal, that
2. calls JS_SetProperty, that
3. calls a proxy method, that
4. calls JS_CallInternal again, that
5. raises an exception, that then
6. calls build_backtrace

The JSStackFrame from (4) has cur_pc set but not the one from (1).

For that particular bytecode handler it's an easy fix of doing `sf->cur_pc = pc` right before the JS_DefineProperty call but we should do a more rigorous review because it's definitely not the only affected bytecode handler.

Reply:
Good catch! I went through a bunch and fixed them. The abort was there to help us fix whatever else we might have missed 😅